utils/forest_shim/src/sector.rs

```diff
@@ -0,0 +1,270 @@
+    SectorInfo as SectorInfoV2, SectorSize as SectorSizeV2,
+};
+use fvm_shared3::sector::{
+    PoStProof as PoStProofV3, RegisteredPoStProof as RegisteredPoStProofV3,
+    RegisteredSealProof as RegisteredSealProofV3, SectorInfo as SectorInfoV3,
+    SectorSize as SectorSizeV3,
+};
+
+use crate::Inner;
+
+/// Represents a shim over RegisteredSealProof from fvm_shared with convenience
+/// methods to convert to an older version of the type
+///
+/// # Examples
+/// ```
+/// 
+/// // Create FVM2 RegisteredSealProof normally
+/// let fvm2_proof = fvm_shared::sector::RegisteredSealProof::StackedDRG2KiBV1;
+///
+/// // Create a correspndoning FVM3 RegisteredSealProof
+/// let fvm3_proof = fvm_shared3::sector::RegisteredSealProof::StackedDRG2KiBV1;
+///
+/// // Create a shim out of fvm2 proof, ensure conversions are correct
+/// let proof_shim = forest_shim::sector::RegisteredSealProof::from(fvm2_proof);
+/// assert_eq!(fvm3_proof, *proof_shim);
+/// assert_eq!(fvm2_proof, proof_shim.into());
+/// ```
+#[derive(serde::Serialize, serde::Deserialize, Clone)]
+pub struct RegisteredSealProof(RegisteredSealProofV3);
+
+impl From<RegisteredSealProofV3> for RegisteredSealProof {
+    fn from(value: RegisteredSealProofV3) -> Self {
+        RegisteredSealProof(value)
+    }
+}
+
+impl crate::Inner for RegisteredSealProof {
+    type FVM = RegisteredSealProofV3;
+}
+
+impl Deref for RegisteredSealProof {
+    type Target = RegisteredSealProofV3;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl From<RegisteredSealProofV2> for RegisteredSealProof {
+    fn from(value: RegisteredSealProofV2) -> RegisteredSealProof {
+        let proof = match value {
+            RegisteredSealProofV2::StackedDRG2KiBV1 => RegisteredSealProofV3::StackedDRG2KiBV1,
+            RegisteredSealProofV2::StackedDRG512MiBV1 => RegisteredSealProofV3::StackedDRG512MiBV1,
+            RegisteredSealProofV2::StackedDRG8MiBV1 => RegisteredSealProofV3::StackedDRG8MiBV1,
+            RegisteredSealProofV2::StackedDRG32GiBV1 => RegisteredSealProofV3::StackedDRG32GiBV1,
+            RegisteredSealProofV2::StackedDRG64GiBV1 => RegisteredSealProofV3::StackedDRG64GiBV1,
+            RegisteredSealProofV2::StackedDRG2KiBV1P1 => RegisteredSealProofV3::StackedDRG2KiBV1P1,
+            RegisteredSealProofV2::StackedDRG512MiBV1P1 => {
+                RegisteredSealProofV3::StackedDRG512MiBV1P1
+            }
+            RegisteredSealProofV2::StackedDRG8MiBV1P1 => RegisteredSealProofV3::StackedDRG8MiBV1P1,
+            RegisteredSealProofV2::StackedDRG32GiBV1P1 => {
+                RegisteredSealProofV3::StackedDRG32GiBV1P1
+            }
+            RegisteredSealProofV2::StackedDRG64GiBV1P1 => {
+                RegisteredSealProofV3::StackedDRG64GiBV1P1
+            }
+            RegisteredSealProofV2::Invalid(i64) => RegisteredSealProofV3::Invalid(i64),
+        };
+        RegisteredSealProof(proof)
```

Comment:
`RegisteredSealProof` can be converted from and to a number. To convert between V2 and V3 (which are identical), you can convert `RegisteredSealProofV2 → i64 → RegisteredSealProofV3`.

Reply:
thanks that's cleaner, RegisteredSealProof and RegisteredPoStProof are now using i64 conversions.